File: leetcode/easy/977-squares-of-a-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    let left = 0;
    let right = nums.length - 1;
    let index = right;
    let result = [];

    while(left <= right) {

        if(Math.abs(nums[left]) >= Math.abs(nums[right])) {
            result[index--] = nums[left] * nums[left];
            left++;
        } else {
            result[index--] = nums[right] * nums[right];
            right--;
        }
    }
    return result;
};

File: leetcode/easy/977-squares-of-a-array.py
from types import List
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums) - 1
        resultIdx = right
        result = [0] * len(nums)

        while left <= right:
            if abs(nums[left]) > abs(nums[right]):
                result[resultIdx] = nums[left] * nums[left]
                left += 1
                resultIdx -= 1
            else:
                result[resultIdx] = nums[right] * nums[right]
                right -= 1
                resultIdx -= 1
        return result
